### PR TITLE
COMMERCE-6422 Fix Order Engine not working with custom order statuses that have the same priority

### PR DESCRIPTION
**Pull request by alecsloan**

### CODE REVIEW

**Comment by alecsloan** [comment]
ci:test:sf

**Comment by alecsloan** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/order/engine/CommerceOrderEngineImpl.java [hunk 1]
```diff
@@ -462,7 +474,7 @@ private boolean _isGuestCheckoutEnabled(long groupId) throws Exception {
 			_configurationProvider.getConfiguration(
 				CommerceOrderCheckoutConfiguration.class,
 				new GroupServiceSettingsLocator(
-					groupId, CommerceConstants.SERVICE_NAME_ORDER));
+					groupId, CommerceConstants.ORDER_SERVICE_NAME));
```
Comment (dsitu) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/order/engine/CommerceOrderEngineImpl.java:477]: Hey @alecsloan we should use `SERVICE_NAME_ORDER`. Its just a rename of  `SERVICE_NAME_ORDER`. `SERVICE_NAME_ORDER` will be deprecated.
```suggestion
					groupId, CommerceConstants.SERVICE_NAME_ORDER));
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Comment by alecsloan** [comment]
ci:test:sf

**Comment by alecsloan** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
@riccardo-alberti can you please take care of this PR?

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Comment by riccardo-alberti** [comment]
ci:forward

**Pull request closed without merging**